### PR TITLE
fix bugs in aggregate wrapper, and emulate count

### PR DESCRIPTION
**Pull request by boutell**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -348,6 +357,78 @@ function decorateCollection(collection) {
...
+      }
+      if (options.skip !== undefined) {
+        cursor.skip(options.skip);
+      }
+      if (options.hint !== undefined) {
+        cursor.hint(options.hint);
+      }
+      const p = cursor.project({ _id: 1 }).toArray().then(function(objects) {
+        if (callback) {
+          callback(null, objects.length);
```
Comment (abea) [inline index.js:394]: Why aren't we returning these callbacks and always returning `p`?

File: test/test.js [hunk 2]
```diff
@@ -162,4 +167,98 @@ describe('use mongodb 3 driver in a 2.x style', function() {
...
+    return trees.aggregate([
+      {
+        $match: {
+          ohmy: true
+        }
+      }
+    ]).then(function(result) {
+      assert(result);
+      assert(Array.isArray(result));
+      assert(result.length);
```
Comment (abea) [inline test/test.js:180]: It seems like we should be testing for specific lengths here. Not a deal breaker, though.

Comment (abea) [inline test/test.js:180]: It seems like we should be testing for specific lengths where we do this assertion. Not a deal breaker, though.

File: index.js (hunk 1 above)
Comment (boutell) [reply index.js:394]: p is a promise. Whether we want to return it depends on whether we were invoked with a callback.

Returning according to our normal "always be returning" pattern would return the promise *and* accept the callback, which leads to mongodb getting mad at us for "over-specifying."

This nonsense only comes up in rare situations like this where we support both promises and callbacks.

File: test/test.js (hunk 2 above)
Comment (boutell) [reply test/test.js:180]: Yeah, it's really not about whether the result is bang on (we know mongo can aggregate) so much as that it smells like we invoked the right things.

**Pull request merged**